pkg/base/helm.go

```diff
@@ -434,116 +435,38 @@ func getAllBasePaths(prefix string, base Base) []string {
 	return basePaths
 }
 
-// insert namespace if it's defined in the spec and not already present in the manifests
-func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
-	if renderOptions.Namespace == "" {
+func insertNamespace(baseFiles []BaseFile, namespace string) ([]BaseFile, error) {
+	if namespace == "" {
 		return baseFiles, nil
 	}
 
-	chartsPath, err := ioutil.TempDir("", "charts")
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to create temp dir")
-	}
-	defer os.RemoveAll(chartsPath)
-
-	var updatedBaseFiles []BaseFile
-	var kustomizeResources []string
-	var kustomizePatches []kustomizetypes.PatchStrategicMerge
-	resources := map[string]BaseFile{}
-	foundGVKNamesMap := map[string]bool{}
-	for _, baseFile := range baseFiles {
-		// write temp files for manifests that need a namespace
-		gvk, manifest := GetGVKWithNameAndNs(baseFile.Content, renderOptions.Namespace)
-		if manifest.APIVersion == "" || manifest.Kind == "" || manifest.Metadata.Name == "" {
-			updatedBaseFiles = append(updatedBaseFiles, baseFile)
-			continue // ignore invalid resources
-		}
-
-		// ignore crds
-		if manifest.Kind == "CustomResourceDefinition" {
-			updatedBaseFiles = append(updatedBaseFiles, baseFile)
+	decoder := serializer.NewCodecFactory(scheme.Scheme).UniversalDeserializer()
+	for i := range baseFiles {
+		obj, gvk, err := decoder.Decode(baseFiles[i].Content, nil, nil)
```

Comment:
what if we don't have the right schema loaded to decode the manifests. were there any issues using the `GetGVKWithNameAndNs` function?

Reply:
Ah, I see you're using this below to inject the namespace. But still, the issue remains.

Reply:
good point.  moved away from decoding objects to simply deleting `nil` map entries.  these were what was causing `null` to get populated in it's place (which I believe is still valid yaml, but Kustomize does not like it)